components/threading/thread_viewer/thread_viewer.tsx

```diff
@@ -0,0 +1,564 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import $ from 'jquery';
+import React, {HTMLAttributes} from 'react';
+import Scrollbars from 'react-custom-scrollbars';
+import classNames from 'classnames';
+
+import {Posts} from 'mattermost-redux/constants';
+import {Channel} from 'mattermost-redux/types/channels';
+import {ExtendedPost} from 'mattermost-redux/actions/posts';
+import {Post} from 'mattermost-redux/types/posts';
+import {UserProfile} from 'mattermost-redux/types/users';
+import {UserThread} from 'mattermost-redux/types/threads';
+
+import Constants from 'utils/constants';
+import DelayedAction from 'utils/delayed_action';
+import * as Utils from 'utils/utils.jsx';
+import * as UserAgent from 'utils/user_agent';
+import CreateComment from 'components/create_comment';
+import DateSeparator from 'components/post_view/date_separator';
+import FloatingTimestamp from 'components/post_view/floating_timestamp';
+import RhsComment from 'components/rhs_comment';
+import RhsRootPost from 'components/rhs_root_post';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import {FakePost} from 'types/store/rhs';
+import {THREADING_TIME as BASE_THREADING_TIME} from 'components/threading/common/options';
+
+import './thread_viewer.scss';
+
+const THREADING_TIME: typeof BASE_THREADING_TIME = {
+    ...BASE_THREADING_TIME,
+    units: [
+        'now',
+        'minute',
+        'hour',
+        'day',
+        'week',
+        'month',
+        'year',
+    ],
+};
+export function renderView(props: Record<string, any>) {
+    return (
```

Comment:
nit: for simple methods like these, I like the reduced-line-count pattern that eliminates the `return`:
```js
export function renderView = (props: Record<string, any>) => ({
    <div ...
});
```

Reply:
(was moved/existing; didn't want to touch unnecessary)

I like that pattern, too. But I'm up on the air on this one. At least while developing, I may add and remove `const`s/`let`s as the function evolves, so switching between these two becomes somewhat burdensome IMHO..  


```tsx
const thing = () => (
    // jsx
);
```
and
```tsx
const thing = () => {
    return (
        // jsx
    );
};
```